app/controllers/opendata/agents/nodes/mypage/app/my_app/appfiles_controller.rb

```diff
@@ -13,130 +13,130 @@ class Opendata::Agents::Nodes::Mypage::App::MyApp::AppfilesController < Applicat
-      {}
-    end
-
-    def permit_fields
-      @model.permitted_fields
-    end
-
-    def get_params
-      params.require(:item).permit(permit_fields).merge(fix_params)
-    end
-
-    def deliver_workflow_mail
-      return unless @route
-      return unless @deliver_mail
-      return unless @item.errors.empty?
-      args = {
-        m_id: @cur_member.id,
-        t_uid: @app.workflow_approvers.first[:user_id],
-        site: @cur_site,
-        item: @app,
-        url: ::File.join(@cur_site.full_url, opendata_app_path(site: @cur_site, cid: @app.parent.id, id: @app))
-      }
-      Opendata::Mailer.request_resource_mail(args).deliver_now rescue nil
-    end
+  def app
+    @app ||= Opendata::App.site(@cur_site).find params[:app_id]
+  end
+
+  def set_app
+    raise "403" if app.member_id != @cur_member.id
+    @app_url     = "#{@cur_node.url}#{@app.id}/"
+    @appfile_url = "#{@app_url}appfiles/"
+  end
+
+  def set_model
+    @model = Opendata::Appfile
+  end
+
+  def set_item
+    @item = app.appfiles.find params[:id]
+    @item_url = "#{@appfile_url}#{@item.id}/"
+  end
+
+  def set_workflow
+    @cur_site = Cms::Site.find(@cur_site.id)
+    @route = @cur_site.app_workflow_route
+  end
+
+  def set_status
+    status = "closed"
+    status = "request" if @route && params[:request].present?
+    status = "public"  if !@route && params[:publish_save].present?
+
+    @item.status = status
+    @item.workflow = { member: @cur_member, route: @route, workflow_reset: true }
+    @deliver_mail = true if status == "request" && @app.status != "request"
+  end
+
+  def fix_params
+    {}
+  end
+
+  def pre_params
+    {}
+  end
+
+  def permit_fields
+    @model.permitted_fields
+  end
+
+  def get_params
+    params.require(:item).permit(permit_fields).merge(fix_params)
+  end
+
+  def deliver_workflow_mail
+    return unless @route
+    return unless @deliver_mail
+    return unless @item.errors.empty?
+    args = {
+      m_id: @cur_member.id,
+      t_uid: @app.workflow_approvers.first[:user_id],
+      site: @cur_site,
+      item: @app,
+      url: ::File.join(@cur_site.full_url, opendata_app_path(site: @cur_site, cid: @app.parent.id, id: @app))
+    }
+    Opendata::Mailer.request_resource_mail(args).deliver_now rescue nil
+  end
 
   public
-    def index
-      @items = @app.appfiles.
-      order_by(name: 1).
-      page(params[:page]).per(50)
+  def index
+    @items = @app.appfiles.
+    order_by(name: 1).
+    page(params[:page]).per(50)
 
-      render
-    end
+    render
+  end
 
-    def show
-      render
-    end
+  def show
+    render
+  end
 
-    def download
-      @item = @app.appfiles.find(params[:appfile_id])
+  def download
+    @item = @app.appfiles.find(params[:appfile_id])
 
-      send_file @item.file.path, type: @item.content_type, filename: @item.filename,
-        disposition: :attachment, x_sendfile: true
-    end
+    send_file @item.file.path, type: @item.content_type, filename: @item.filename,
```

Comment:
__[brakeman]__
* Parameter value used in file name [:ref](http://brakemanscanner.org/docs/warning_types/file_access/)

<!-- Comment by [SideCI](https://sideci.com) -->